ff/src/fields/bf.rs

```diff
@@ -0,0 +1,153 @@
+use num_bigint::BigUint;
+use num_traits::{Zero, One};
+
+#[derive( Clone, PartialEq, Eq)]
+pub struct BinaryFieldElement {
+    value: BigUint,
+}
```

Comment:
Here I'm not sure this way of doing things is the more efficient approach because no matter the size of the field concerned, you store in your `value` as `BigUint` a vector of `BigDigit` (u64 or 32 I think) but for field extensions before 32, we don't need as much space.

So I wonder if it is not more optimized to do something custom made like this:

https://gitlab.com/IrreducibleOSS/binius/-/blob/main/crates/field/src/binary_field.rs?ref_type=heads#L770-L777

so that we can have a very efficient implementation for small extensions.

Reply:
This makes a lot of sense, but it would mean implementing a separate Binary Field implementation for each field size. Should I submit these changes as part of my next PR, or should I first push the other changes we discussed (operators and moving binmul) and then implement the different field sizes in subsequent PRs?